frontends/p4/sideEffects.cpp

```diff
@@ -51,9 +51,10 @@ struct EvaluationOrder {
     }
 
     const IR::Expression* addAssignment(
+        Util::SourceInfo srcInfo,
         cstring varName, const IR::Expression* expression) {
         auto left = new IR::PathExpression(IR::ID(varName, nullptr));
-        auto stat = new IR::AssignmentStatement(left, expression);
+        auto stat = new IR::AssignmentStatement(srcInfo, left, expression);
```

Comment:
@mbudiu-vmw are you sure that these are related to issue #532? The seem to me more appropriate for https://github.com/p4lang/p4c/pull/509